packages/hash/frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-editor/types-section.tsx

```diff
@@ -1,59 +1,133 @@
+import { EntityId } from "@hashintel/hash-shared/types";
+import { versionedUriFromComponents } from "@hashintel/hash-subgraph/src/shared/type-system-patch";
 import { getEntityTypeById } from "@hashintel/hash-subgraph/src/stdlib/element/entity-type";
 import { getRoots } from "@hashintel/hash-subgraph/src/stdlib/roots";
-import { Box, Typography } from "@mui/material";
+import { Box } from "@mui/material";
+import { useEffect, useState } from "react";
 
+import { useBlockProtocolUpdateEntity } from "../../../../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-update-entity";
+import { useBlockProtocolAggregateEntityTypes } from "../../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-aggregate-entity-types";
 import { SectionWrapper } from "../../../shared/section-wrapper";
-import { WhiteCard } from "../../../shared/white-card";
 import { useEntityEditor } from "./entity-editor-context";
-
-interface TypeCardProps {
-  url: string;
-  title: string;
-}
-
-const TypeCard = ({ url, title }: TypeCardProps) => {
-  return (
-    <WhiteCard href={url}>
-      <Box
-        sx={{
-          display: "flex",
-          alignItems: "center",
-          px: 1.5,
-          py: 1.25,
-          gap: 1.25,
-          color: ({ palette }) => palette.black,
-        }}
-      >
-        <FontAwesomeIcon icon={faAsterisk} />
-        <Typography variant="smallTextLabels" fontWeight={600}>
-          {title}
-        </Typography>
-      </Box>
-    </WhiteCard>
-  );
-};
+import { EntityTypeUpdateModal } from "./types-section/entity-type-update-modal";
+import { TypeCard } from "./types-section/type-card";
 
 export const TypesSection = () => {
-  const { entitySubgraph } = useEntityEditor();
+  const { entitySubgraph, refetch } = useEntityEditor();
 
   const entity = getRoots(entitySubgraph)[0]!;
+  const { updateEntity } = useBlockProtocolUpdateEntity();
+  const {
+    metadata: { editionId, entityTypeId },
+    properties,
+  } = entity;
+
+  const { aggregateEntityTypes } = useBlockProtocolAggregateEntityTypes();
+  const [newVersion, setNewVersion] = useState<number>();
+  const [updateModalOpen, setUpdateModalOpen] = useState(false);
+  const [updatingVersion, setUpdatingVersion] = useState(false);
+
+  useEffect(() => {
+    const init = async () => {
+      /** @todo instead of aggregating all types, use filtering by baseId when it's available to use */
+      const res = await aggregateEntityTypes({
+        data: {
+          graphResolveDepths: {
+            constrainsValuesOn: { outgoing: 0 },
+            constrainsPropertiesOn: { outgoing: 0 },
+            constrainsLinksOn: { outgoing: 0 },
+            constrainsLinkDestinationsOn: { outgoing: 0 },
+          },
+        },
+      });
+
+      const baseId = extractBaseUri(entityTypeId);
+      const entityTypeWithSameBaseId = res.data?.roots.find(
+        (root) => root.baseId === baseId,
+      );
```

Comment:
What happens if there are 100 versions of the same type? Is the first one guaranteed to be the latest?

Reply:
`aggregateEntityTypes` returns just one item for each entity type. And gives it's latest version number as well. (at least based on my tests, did not read the code, but tried it multiple times.)

Reply:
Ah right, got it, thanks!